libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -50,6 +69,62 @@ public static void initialize(Instrumentation inst) throws Exception {
         inst.retransformClasses(System.class);
     }
 
+    private static PolicyManager createPolicyManager() throws IOException {
+        Map<String, Policy> pluginPolicies = createPluginPolicies();
+
+        // TODO: What should the name be?
+        // TODO(ES-10031): Decide what goes in the elasticsearch default policy and extend it
+        var elasticsearchPolicy = new Policy("elasticsearch", List.of());
+        return new PolicyManager(elasticsearchPolicy, pluginPolicies, EntitlementBootstrap.pluginResolver);
+    }
+
+    private static Map<String, Policy> createPluginPolicies() throws IOException {
+        Map<String, Policy> pluginPolicies = new HashMap<>(EntitlementBootstrap.pluginData.size());
```

Comment:
Nit: can we pull the args at the top level in `initialize` and then pass them around here? ie don't reach out from within these methods, but pass them in so the inputs are clear. It would also make unit testing these easier.